dbt/parser.py

```diff
@@ -181,15 +183,13 @@ def parse_macro_file(macro_file_path,
 
 
 def parse_node(node, node_path, root_project_config, package_project_config,
-               all_projects, tags=None, fqn_extra=None, fqn=None):
+               all_projects, tags=None, fqn_extra=None, fqn=None, macros=None):
     logger.debug("Parsing {}".format(node_path))
     node = copy.deepcopy(node)
 
-    if tags is None:
-        tags = set()
-
-    if fqn_extra is None:
-        fqn_extra = []
+    tags = coalesce(tags, set())
+    fqn_extra = coalesce(fqn_extra, [])
+    macros = coalesce(macros, {})
```

Comment:
i like this

Reply:
i intend to rewrite dbt-core in sql, and this is step 1 of 10 billion

i think this could equivalently be:
```python
tags = tags || set()
```

but i always find that confusing, and I think codeclimate would still complain about cyclomatic complexity. :+1: